### DISCUSSION TITLE
How make a multi-line string to a single-line space separated string

### DISCUSSION THREAD

**Discussion by LinArcX** [nushell/nushell#7795] [Q&A] [answered]
Imagine that I have a string like this:

```
let test = '''
  foo
  bar
  zoo
'''
```

If I print it, it will show me:
```
''
foo
bar
zoo
''
```
But I want to have this output: `foo bar zoo`

**Comment by fdncred** [comment]
You'd want to do something like `$test | str join`. See `str join --help` for more information.

**Comment by LinArcX** [comment]
Thanks for the reply, but the output didn't change, they are still in different lines:

```
〉$test | str join                                      
''
foo
bar
zoo
''
```

**Comment by fdncred** [comment, marked as answer]
your python quote strings are messing things up. a triple single quote isn't a thing in nushell.

The way to make this sort of thing work in nushell is to have a list to begin with in `$test`. So, something like
```
let test = [foo bar zoo]
```
Then you can join them with any character or no character with 
```
$test | str join
foobarzoo
```
or
```
$test | str join ' '
foo bar zoo
```
if you really need the triple single quote, you'd have to do something like this
```
let test = ["'''" foo bar zoo "'''"]
$test
╭───┬─────╮
│ 0 │ ''' │
│ 1 │ foo │
│ 2 │ bar │
│ 3 │ zoo │
│ 4 │ ''' │
╰───┴─────╯
```

**Reply by LinArcX** [reply]
I don't want python quote strings. I just want to have a string that each line of it contains a word. I don't know what's the best way to achieve that in the Nushell.

And after that, I want to flat them as: `foo bar zoo`

**Reply by fdncred** [reply]
These are called lists in nushell.
```
let test = [foo bar zoo]
$test
╭───┬─────╮
│ 0 │ foo │
│ 1 │ bar │
│ 2 │ zoo │
╰───┴─────╯
$test | str join ' '
foo bar zoo
```

**Reply by LinArcX** [reply]
Yes, I should change my mindset about strings. I think lists are a great way to handle such things.
